Modules/xSharePoint/DSCResources/MSFT_xSPSearchIndexPartition/MSFT_xSPSearchIndexPartition.psm1

```diff
@@ -0,0 +1,170 @@
+function Get-TargetResource
+{
+    [CmdletBinding()]
+    [OutputType([System.Collections.Hashtable])]
+    param
+    (
+        [parameter(Mandatory = $true)]  [System.UInt32]    $Index,
+        [parameter(Mandatory = $true)]  [System.String[]]  $Servers,
+        [parameter(Mandatory = $false)] [System.String]    $RootDirectory,
+        [parameter(Mandatory = $true)]  [System.String]    $ServiceAppName,
+        [parameter(Mandatory = $false)] [System.Management.Automation.PSCredential] $InstallAccount
+    )
+
+    $result = Invoke-xSharePointCommand -Credential $InstallAccount -Arguments $PSBoundParameters -ScriptBlock {
+        $params = $args[0]
+        $ConfirmPreference = 'None'
+
+        $ssa = Get-SPEnterpriseSearchServiceApplication -Identity $params.ServiceAppName      
+        $currentTopology = $ssa.ActiveTopology
+        
+        $IndexComponents = (Get-SPEnterpriseSearchComponent -SearchTopology $currentTopology | Where-Object { ($_.GetType().Name -eq "IndexComponent") -and ($_.IndexPartitionOrdinal -eq $params.Index) }).ServerName
```

Comment:
Nope, I've used this in my labs to provision out a brand new service app, it works fine where there is no topology. Provisioned another one out from scratch last night.